src/internal/pachsql/db_test.go

```diff
@@ -0,0 +1,167 @@
+	schemaName  = "schemaname"
+	tableName   = "tablename"
+)
+
+var (
+	commitsTable     = &SchemaTable{collections, commits}
+	commitDiffsTable = &SchemaTable{pfs, commitDiffs}
+)
+
+func TestListTables(t *testing.T) {
+	t.Run("ListTables() with a valid response", listTablesValidResponse)
+	t.Run("ListTables() with an unexpected response", listTablesUnexpectedResponse)
+	t.Run("ListTables() with an error from the database", listTablesInvalidResponse)
+}
+
+func listTablesValidResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{schemaName, tableName}).
+		AddRow(collections, commits).
+		AddRow(pfs, commitDiffs)
+	mock.ExpectQuery(listTablesQuery).WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	schemaTables, err := ListTables(context.Background(), sqlxDB)
+	require.NoError(t, err, "test should pass")
+	require.Equal(t, 2, len(schemaTables),
+		"ListTables() should return same number of rows as input")
+}
+
+func listTablesUnexpectedResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{"food"}).
+		AddRow("sushi").
+		AddRow("cheeseburger")
+	mock.ExpectQuery(listTablesQuery).WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	_, err = ListTables(context.Background(), sqlxDB)
+	require.Equal(t, true,
+		err != nil && strings.HasPrefix(err.Error(), "could not scan tables:"),
+		"expected an error from ListTables()")
+}
+
+func listTablesInvalidResponse(t *testing.T) {
+	mockDB, _, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	_, err = ListTables(context.Background(), sqlxDB)
+	require.Equal(t, true, err != nil && strings.HasPrefix(err.Error(), "could not list tables:"),
+		"expected an error from ListTables()")
+}
+
+func TestScannedRows(t *testing.T) {
+	t.Run("ScannedRows() with a valid response", scannedRowsValidResponse)
+	t.Run("ScannedRows() with a valid response", scannedRowsInvalidResponse)
+}
+
+func scannedRowsValidResponse(t *testing.T) {
+	mockDB, mock, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	mockRows := sqlmock.NewRows([]string{schemaName, tableName}).
+		AddRow(collections, commits).
+		AddRow(pfs, commitDiffs)
+	mock.ExpectQuery("").WillReturnRows(mockRows)
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	rows, err := ScannedRows(context.Background(), sqlxDB, "")
+	require.NoError(t, err)
+	require.Equal(t, 2, len(rows))
+	require.Equal(t, rows[0][schemaName], "collections")
+}
+
+func scannedRowsInvalidResponse(t *testing.T) {
+	mockDB, _, err := sqlmock.New()
+	require.NoError(t, err)
+	defer mockDB.Close()
+	sqlxDB := sqlx.NewDb(mockDB, mockDBName)
+	_, err = ScannedRows(context.Background(), sqlxDB, "")
+	require.Equal(t, true, err != nil && strings.HasPrefix(err.Error(), "failed to query database:"),
```

Comment:
There is no way to debug this test if it fails.  It will print "false != true" when you really want it to print the actual error message that was returned.

I tend to do this: https://github.com/jrockway/json-logs/blob/master/pkg/parse/parse_test.go#L969.  Basically, `cmp.Diff` on the received error vs a special type that matches a regex on the error (or an exact error if you want).

For this codebase, I'd require.YesError, and then require.Matches, though.